### ISSUE TITLE
Conversion issue

### ISSUE THREAD

**Issue by GCamp6** [kedder/ofxstatement#57]
After requesting ISO20022 account files I received xml file (and not csv). I installed postfinace and ISO20022 plugin and after running

`ofxstatement convert -t pf-bank-tab input.xml output.ofx`

I see on the terminal the reading of the file but at the end I get: 

> Document> does not match

Is it possible to include a conversion from xml to csv(if that's the mistake)?

I tried to run the iso20022 plugin on the xml file but I get:

> AttributeError: 'NoneType' object has no attribute 'text'

I managed to transform the xml into csv but still, after running
`ofxstatement convert -t pf-bank-tab input.csv output.ofx`

I get
> Exception: "all the first line definitions divided by ," does not match

**Comment by kedder** [comment]
@GCamp6, thanks for the report. 

To solve the problem, I need some way to reproduce it. Can you provide some sort of anonymized input file? If you don't want to expose your personal information, you can manually remove it from the file you have. 

Is your file actually in ISO-20022 format? ISO-20022 files looks like

```
<?xml version="1.0" encoding="utf-8"?>
<Document xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="urn:iso:std:iso:20022:tech:xsd:camt.053.001.02">
  <BkToCstmrStmt>
    <GrpHdr>
      <MsgId>165999526</MsgId>
      <CreDtTm>2016-01-01T18:40:37</CreDtTm>
    </GrpHdr>
    <Stmt>
      <Id>LT000000000000000000EUR000000000</Id>
...
```

**Comment by GCamp6** [comment]
I asked on purpose the ISO20022 standard format and the bank sent me the xls file (here I can not upload xml, how can I add code text without filling all the page?)

the first lines are:

`<?xml version="1.0" encoding="UTF-8"?>
<Document xmlns="urn:iso:std:iso:20022:tech:xsd:camt.053.001.02" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<BkToCstmrStmt>
		<GrpHdr>
			<MsgId>KOBE170214357689</MsgId>
			<CreDtTm>2017-02-14T10:30:00.00Z</CreDtTm>
			<MsgPgntn>
				<PgNb>1</PgNb>
				<LastPgInd>true</LastPgInd>
			</MsgPgntn>
		</GrpHdr>
		<Stmt>
			<Id>KOBE170214357689</Id>
			<ElctrncSeqNb>999</ElctrncSeqNb>
			<CreDtTm>2017-02-14T10:30:00.00Z</CreDtTm>
			<CpyDplctInd>DUPL</CpyDplctInd>
			<Acct>
				<Id>
					<IBAN>IBAN</IBAN>
				</Id>
				<Ownr>
					<Nm>Surname Name</Nm>
				</Ownr>
			</Acct>`

**Comment by kedder** [comment]
You can paste the whole file on some paste board (e.g. https://dpaste.de/) and post a link it here.

**Comment by GCamp6** [comment]
That's my polished file:

[LinkAccount.xml](https://mega.nz/#!rUlx0LDS!CFNcg7OFHWH_tL92lfuSAJ4qQFEOxTDvD4fUylxQflI)

**Comment by GCamp6** [comment]
Any bug discovered?

**Comment by kedder** [comment]
@GCamp6, sorry for delay. I was able to reproduce the problem and fix it. [ofxstatement-iso20022](https://pypi.python.org/pypi/ofxstatement-iso20022/0.2.0)-2.0 was released.

Please update `ofxstatement-iso20022` plugin and try on your real file. `pip3 install ofxstatement-iso20022 --upgrade` should work.

Let me know if something is not right.

**Comment by GCamp6** [comment]
Thanks a lot!
Now the conversion process is working.

Still some issues:

- when I import the new ofx file I see only a fewer number of items (5 out of 362 of the original xml file or 5 out of 7)
- the "amount" money column during the import procedure is empty (0.00)

**Comment by kedder** [comment]
Hm, interesting. I'll look closer.

**Comment by GCamp6** [comment]
Strange thing:
I tried to convert the same xml file with ofxstatement-iso2022-0.2.0 but from my linux machine (mint 18.1) and not windows as previously. Everything ok but then, if I import the new ofx file into Gnucash, now I find that the number of item is correct (?!).
(at the end of the ofxstatement-iso20022 update process the outcome from the terminal was positive in both pc).

The dimensions of the ofx files converted from the different pc is the same.

Still the amount of money is at 0.00

**Comment by kedder** [comment]
Ok, I figured out what's wrong: the statement has no "currency" specified on account (even though there is a currency attribute on every statement line (assuming there can be separate currency for each line).

Since GnuCash doesn't work well with multi-currency statements (and you have to specify an account with defined currency when you import OFX file), the current approach is to generate a separate OFX file for each currency in a statement. For this we can support a configuration option. 

For example I use two "types" to parse the single statement - for EUR and USD. My config looks like this:
```
[swedbank-eur]                                                                   
plugin = swedbank                                                                
currency = EUR                                                                   
                                                                                 
[swedbank-usd]                                                                   
plugin = swedbank                                                                
currency = USD
``` 
I get two ofx files and import them into GnuCash separately.

Would that solution work for you? You would have to specify currency explicitly in config.

**Comment by GCamp6** [comment]
I see.
Actually, just to clarify, the xml file I uploaded had only one currency (CHF) and I tried to import it into a CHF account in GnuCash. In my GnuCash file there are both € and CHF accounts.

I assume having two plugins making the job wouldn't be an issue.

_PS: At the beginning I was using the postfinance pf-bank-tab plugin (my bank xml file) but did not work. Do you know why? Wouldn't it be the best option?_

**Comment by kedder** [comment]
I just uploaded version 0.3.1 of iso20022 plugin. I believe you should be able to import your statement now. To do this, edit your configuration with `ofxstatement edit-config` and put something like this there:

```
[yourbank:chf]                                                                            
plugin = iso20022                                                                
currency = CHF  

[yourbank:eur]                                                                            
plugin = iso20022                                                                
currency = CHF  
```

You can convert your CHF and EUR statements with these commands:

```
ofxstatement convert -t yourbank:chf input_chf.xml output_chf.ofx
ofxstatement convert -t yourbank:eur input_eur.xml output_eur.ofx
```

Make sure the currency of GnuCash account you are importing into matches currency of the OFX statement.  

Let me know if something is still wrong.

**Comment by GCamp6** [comment]
So I downloaded and updated the plugin.
(running from the terminal `pip3 install ofxstatement-iso20022 --upgrade` I get an error 
> AttributeError: '_NamespacePath' object has no attribute 'sort'

)

With `ofxstatement --version` I get

> ofxstatement 0.6.0


I add with `ofxstatement edit-config`

[pf:chf]
plugin=iso20022
currency = CHF  

[pf:eur]
plugin=iso20022
currency = EUR

and I have the config.ini file in the .config folder.

After running: 
ofxstatement convert -t pf:chf input_chf.xml output_chf.ofx
ofxstatement convert -t pf:eur input_eur.xml output_eur.ofx

I have the two .ofx files but in the import procedure in GnuCash still all the values are 0.00

**Comment by kedder** [comment]
Ok, the first issue seems to be similar to #55. Can you run `pip3 help` and show me contents of `ofxstatement-0.6.0-py3.5-nspkg.pth` file referenced in the error. If you can, please add comment on #55.

For the second issue: we need to make sure the file you are importing into GnuCash and account you select in GnuCash has the same currency. Does GnuCash offer you to select an account before you see "import" window with 0.00 in it?

**Comment by GCamp6** [comment]
Sorry for the delay - I had a problem with the pc and I reinstalled the OS. Now I also re-installed pip3 and python3 and repeated the procedure to configure ofxstatement.

I managed to convert .xml and import the new .ofx file without problem and seems to be perfect. "Unfortunately" I solved the error without figuring out what the issue was. Apparently a problem related with wrong/incomplete python installation...
